lang/java/reef-io/src/main/java/org/apache/reef/io/data/output/OutputStreamProvider.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.reef.io.data.output;
+
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+/**
+ * A provider through which users create output streams.
+ */
+public interface OutputStreamProvider {
+
+  /**
+   * create an output stream using the given name.
+   *
+   * @param name name of the created output stream
+   *             It is used as the name of the directory if the created output stream is a file output stream
+   * @return created output stream
+   * @throws java.io.IOException
+   */
+  DataOutputStream create(final String name) throws IOException;
+
+  /**
+   * set the id of the current task.
+   *
+   * @param taskId id of the current task
+   */
+  void setTaskId(final String taskId);
```

Comment:
I agree with your opinion that task ID related methods should be hided from user codes.
I changed the methods' access modifiers to `protected`, and I also extracted the methods using abstract class since they are commonly used.